src/Microsoft.AspNet.OData.Shared/ResourceContext.cs

```diff
@@ -240,6 +240,13 @@ private object BuildResourceInstance()
                 object value;
                 if (EdmObject.TryGetPropertyValue(property.Name, out value) && value != null)
                 {
+                    SelectExpandWrapper valueWrapper = value as SelectExpandWrapper;
+                    if (valueWrapper != null)
+                    {
+                        // Skip the select expand property
+                        continue;
+                    }
+
```

Comment:
or just:

if (value is SelectExpandWrapper)
{
    // skip the select expand property
    continue;
}

Reply:
> would be good to add tests covering this case. Also, suggested a small code simplification, but otherwise lgtm.

Added.